### PR TITLE
Sam to fastq

### PR DESCRIPTION
**Pull request by drosofff**
Just in case we need it independently of bowtie2

### CODE REVIEW

File: tools/sam_to_fastq/sam_to_fastq.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        '--input', action="store", type=str, help="input SAM file")
+    the_parser.add_argument(
+        '--output', action="store", type=str, help="output FASTQ file")
+    args = the_parser.parse_args()
+    return args
+    
+        
+def print_fastq_sequence(samline, file):
+  samfields = samline[:-1].split("\t")
+  print >> file, '%s\n%s\n%s\n%s' % (samfields[0], samfields[9], samfields[0], samfields[10])
```
Comment (mvdbeek) [inline tools/sam_to_fastq/sam_to_fastq.py]: can we go with the common python2/3 approach?
```
file.write('%s\n%s\n%s\n%s' % (samfields[0], samfields[9], samfields[0], samfields[10]))
```

File: tools/sam_to_fastq/sam_to_fastq.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+def main(input, output):
+    infile = open (input, "r")
```
Comment (mvdbeek) [inline tools/sam_to_fastq/sam_to_fastq.py]:
```
with open(input, "r") as infile:
   with open(output, "w") as outfile:
```

would be best-practice.

File: tools/sam_to_fastq/sam_to_fastq.xml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+       <param name="input" type="data" format="sam" label="SAM file for extraction of aligned reads in FASTQ format"/>
+   </inputs>
+
+ <outputs>
+   <data format="fastqsanger" name="output" label="FASTQ extraction" />
+</outputs>
+
+    <tests>
+        <test>
+            <param ftype="fasta" name="input" value="input.sam" />
```
Comment (mvdbeek) [inline tools/sam_to_fastq/sam_to_fastq.xml]: ftype should be "sam", right?

**Comment by mvdbeek** [comment]
yes, cool. I can see how this can be useful!

File: tools/sam_to_fastq/sam_to_fastq.py (hunk 1 above)
Comment (drosofff) [inline tools/sam_to_fastq/sam_to_fastq.py]: yes, and I will also correct bad fastq header generated by this code

**Comment by mvdbeek** [comment]
will this also work for paired end?

**Comment by drosofff** [comment]
Well, I guess it depends on the purpose. I will see this later. Metavisitor
does not use paired ended information and treat paired end as single, for
the moment.

Christophe Antoniewski

ARTbio <http://artbio.fr/> - Tel +33 1 44 2*7 70 05*
Drosophila Genetics & Epigenetics <http://drosophile.org/> - Tel +33 1 44 2
*7 34 39*Mobile +33 6 68 60 51 50
https://twitter.com/ARTbio_IBPS
https://twitter.com/drosofff

2016-03-21 22:24 GMT+01:00 Marius van den Beek <notifications@github.com>:

**Comment by mvdbeek** [comment]
:+1:

**Pull request merged**